### PR TITLE
Feature/product details

### PR DESCRIPTION
**Pull request by dmitrii-casapiuc**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -10,6 +10,8 @@ services:
             MYSQL_DATABASE: ${MYSQL_DATABASE}
             MYSQL_USER: ${MYSQL_USER}
             MYSQL_PASSWORD: ${MYSQL_PASSWORD}
+        ports:
+            - 3306:3306
```
Comment (WictorT) [inline docker-compose.yml]: remove

File: node/src/App.js [hunk 2]
```diff
@@ -1,18 +1,27 @@
...
 class App extends Component {
 
     render() {
 
-        return (
-            <React.Fragment>
-                <Header />
-                <Lists />
-            </React.Fragment>
+        let routes = (
```
Comment (WictorT) [inline node/src/App.js]: do not extract to variable here

**Review by WictorT — CHANGES_REQUESTED** [review]

File: node/src/app/ProductDetails/ProductDetails.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        this.props.fetchProductById(this.props.match.params.id)
+    }
+
+    render() {
+        const {classes, product, loading} = this.props
+
+        return (
+            <React.Fragment>
+            {
+                loading || !product
```
Comment (WictorT) [inline node/src/app/ProductDetails/ProductDetails.js]: rearange code

File: node/src/app/ProductDetails/ProductDetails.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
+import React, {Component} from "react"
```
Comment (WictorT) [inline node/src/app/ProductDetails/ProductDetails.js:1]: Rename directories to lowercase everywhere

File: node/src/shared/store/actions/products.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import {FETCH_PRODUCTS_SUCCESS, FETCH_PRODUCT_SUCCESS} from "./actionsTypes"
```
Comment (WictorT) [inline node/src/shared/store/actions/products.js:1]: shared  folder is used for components that are used in multiple places. It does not seem it to be one of this kind

File: node/src/shared/store/actions/products.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+export function fetchProductsSuccess(products) {
+    return {
+        type: FETCH_PRODUCTS_SUCCESS,
+        products
+    }
+}
+
+export function fetchProductById(productId) {
+    return async dispatch => {
+        const host = process.env.REACT_APP_HOST
```
Comment (WictorT) [inline node/src/shared/store/actions/products.js:23]: rename this variable and include `/api/` in the link

**Review by WictorT — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (dmitrii-casapiuc) [reply docker-compose.yml]: Done

File: node/src/App.js (hunk 2 above)
Comment (dmitrii-casapiuc) [reply node/src/App.js]: Done

File: node/src/app/ProductDetails/ProductDetails.js (hunk 3 above)
Comment (dmitrii-casapiuc) [reply node/src/app/ProductDetails/ProductDetails.js]: Done

File: node/src/shared/store/actions/products.js (hunk 5 above)
Comment (dmitrii-casapiuc) [reply node/src/shared/store/actions/products.js:1]: Done

**Review by WictorT — APPROVED** [review]

**Pull request merged**